### DISCUSSION TITLE
Maximum message size when using websockets

### DISCUSSION THREAD

**Discussion by simoneMasm** [mqttjs/MQTT.js#1606] [Q&A]
Hi all,
I am using MQTT.js v4.3.7, with mosquitto v1.4.11
 as broker and MQTT 3 enabled.  I have a NodeJS (v16.15.1) project in which I connect to the broker using **websockets** as connection protocol.

If the Node MQTT Client sends a message with a size greater than 29kb it works fine. On the contrary, when an other client sends it a message greater than 29 kb it is not received (I know the topic is correct since a message less than 29kb is received) and it may even crash sometimes.
I suspect there is some limitation on the websoscket connection or some options I may be missing since if I change the protocol to mqtt the message is received correctly. Unfortunatly, I am forced to use websockets for project reasons.

Any suggestion?

Thank you.

![Screenshot_1](https://github.com/mqttjs/MQTT.js/assets/133883765/4178c769-0cfd-48c0-abce-173f4267cc75)

**Comment by jsiegenthaler** [comment]
Wouldn't the first thing be to upgrade Node to the current LTS version?
You really want to make sure your platform and all dependencies are up to date before hunting for weird problems...

**Comment by simoneMasm** [comment]
Upgrading mosquitto from version 1.4.11 to 1.5.7 and libwebsockets v1.2 (mosquitto dependency) to v1.5 solved the problem
